### PR TITLE
fix: #4758 added EvaluatedValuePopup over PrimaryColumnsControl to show errors and handled duplicate column name

### LINKED ISSUE #4758: [Enhancement] Avoid duplicating table column header
## Summary

Currently user can have duplicated table column headers which can create confusion  

## Requirement
| Title | Desc | Comment | 
| ----- | ----- | ------- | 
| Highlight Duplicate | If a duplicate tile exist highlight the tile with a red(from theme) border | - |
| Show error | Show the error in the evaluated value popup and in the console too | - |

### PR DESCRIPTION
**Pull request by vibhandikyash**
## Description

> Handled Duplicate Column names

Fixes #4758 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Tested locally.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/4758-duplicate-table-column-header 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 54.59 **(-0.07)** | 36.29 **(-0.06)** | 33.08 **(-0.03)** | 55.1 **(-0.08)**
 :red_circle: | app/client/src/components/ads/DraggableListComponent.tsx | 42.42 **(-1.33)** | 50 **(0)** | 37.5 **(0)** | 44.44 **(-1.71)**
 :red_circle: | app/client/src/components/propertyControls/PrimaryColumnsControl.tsx | 19.44 **(-0.93)** | 6.67 **(-3.01)** | 8.82 **(0.49)** | 19.76 **(-1.89)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.04 **(-0.22)** | 40.42 **(-0.83)** | 34.48 **(0)** | 55.96 **(-0.26)**
 :red_circle: | app/client/src/widgets/TableWidget/widget/propertyUtils.ts | 24.43 **(-0.77)** | 8 **(-0.33)** | 20 **(-3.08)** | 22.95 **(-0.73)**</details>

### CODE REVIEW

**Comment by rahulramesha** [comment]
@vibhandikyash Code looks good to me. Not able to Verify since Deploy preview is not built, could you please merge latest from release.
Also could you add Cypress tests for the scenarios.

**Review by rahulramesha — CHANGES_REQUESTED** [review]
please address the above comment

**Comment by vibhandikyash** [comment]
https://github.com/appsmithorg/appsmith/actions/runs/1442923837

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/Table_Duplicate_ColumnName_spec.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    cy.tableColumnPopertyUpdate("email", "TestUpdated");
+    cy.wait("@updateLayout");
+    cy.get(commonlocators.evaluatedTypeTitle).should("exist");
+
+    // Updating the column name ; "userName" > "TestUpdated2"
+    // this will move focus of input to another column input and let popup close
+    cy.tableColumnPopertyUpdate("userName", "TestUpdated2");
+
+    // duplicate column's border should remain red
+    cy.get("[data-rbd-draggable-id='email'] div[value='TestUpdated']").should(
```
Comment (rahulramesha) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/Table_Duplicate_ColumnName_spec.js:24]: nit: The CSS color can be changed in the future, that will fail this test. any other indicator to verify this?

**Review by rahulramesha — APPROVED** [review]

**Review by rahulramesha — APPROVED** [review]

**Comment by vibhandikyash** [comment]
https://github.com/appsmithorg/appsmith/actions/runs/1448351592

**Pull request merged**